server/test/otp.test.js

```diff
@@ -89,8 +85,159 @@ test('/otp route', async (t) => {
     t.equal(response.statusCode, 403)
     t.equal(docStub.calledThrice, true)
     t.equal(sendStub.called, true)
+    clock.restore()
   })
 
+  t.test(
+    'should generate push notification on POST request with valid body',
+    async (t) => {
+      const clock = sinon.useFakeTimers()
+      // All tokens collection
+      docStub.onFirstCall().returns({
+        get: () => ({
+          exists: true,
+          data: () => ({
+            secretId: 'secretId',
+            subscriptionId: 'subscriptionId'
+          })
+        })
+      })
+      // Subscriptions collection
+      docStub.onSecondCall().returns({
+        get: () => ({
+          exists: true,
+          data: () => sinon.stub()
+        })
+      })
+      // Requests collection
+      docStub.onThirdCall().returns({
+        set: () => {},
+        onSnapshot: () => sinon.stub(),
+        delete: () => sinon.stub()
+      })
+
+      let response
+
+      server
+        .inject({
+          url: '/api/generate/55555',
+          method: 'POST',
+          body: {
+            packageInfo: {
+              version: 'v2',
+              name: '@optic/optic-expo'
+            }
+          }
+        })
+        .then((resp) => (response = resp))
+
+      await clock.tickAsync(61e3)
+
+      t.equal(response.statusCode, 403)
+      t.equal(docStub.calledThrice, true)
+      t.equal(sendStub.called, true)
+      clock.restore()
+    }
+  )
+
+  t.test(
+    'should generate push notification on POST request without body',
+    async (t) => {
+      const clock = sinon.useFakeTimers()
+      // All tokens collection
+      docStub.onFirstCall().returns({
+        get: () => ({
+          exists: true,
+          data: () => ({
+            secretId: 'secretId',
+            subscriptionId: 'subscriptionId'
+          })
+        })
+      })
+      // Subscriptions collection
+      docStub.onSecondCall().returns({
+        get: () => ({
+          exists: true,
+          data: () => sinon.stub()
+        })
+      })
+      // Requests collection
+      docStub.onThirdCall().returns({
+        set: () => {},
+        onSnapshot: () => sinon.stub(),
+        delete: () => sinon.stub()
+      })
+
+      let response
+
+      server
+        .inject({
+          url: '/api/generate/55555',
+          method: 'POST'
+        })
+        .then((resp) => (response = resp))
+
+      await clock.tickAsync(61e3)
+
+      t.equal(response.statusCode, 403)
+      t.equal(docStub.calledThrice, true)
+      t.equal(sendStub.called, true)
+      clock.restore()
```

Comment:
this is not a good pattern because if one test fails, the timers will remain faked. I see two options:

- either to configure the fake timers in setup/teardown functions which apply to all tests
- you restore the clock in each specific test but in a teardown function, in order to guarantee that it will be executed